newIDE/app/src/ResourcesList/ResourcePreview/index.js

```diff
@@ -16,80 +15,50 @@ type Props = {|
   onSize?: (number, number) => void,
 |};
 
-type State = {|
-  resourceKind: ?ResourceKind,
-|};
-
 /**
  * Display the right preview for any given resource of a project
  */
-export default class ResourcePreview extends React.PureComponent<Props, State> {
-  state = this._loadFrom(this.props);
+const ResourcePreview = (props: Props) => {
+  const { project, resourceName } = props;
+  const resourcesManager = project.getResourcesManager();
+  const resourceKind = resourcesManager.hasResource(resourceName)
+    ? resourcesManager.getResource(resourceName).getKind()
+    : null;
 
-  // To be updated, see https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops.
-  UNSAFE_componentWillReceiveProps(newProps: Props) {
-    if (
-      newProps.resourceName !== this.props.resourceName ||
-      newProps.project !== this.props.project
-    ) {
-      this.setState(this._loadFrom(newProps));
-    }
+  switch (resourceKind) {
+    case 'image':
+      return (
+        <ImagePreview
+          resourceName={props.resourceName}
```

Comment:
seems like you can replace props.resourceName & props.project with `resourceName` & `project` everywhere below